pkg/plugins/plugins.go

```diff
@@ -41,26 +41,33 @@ var (
 )
 
 type PluginScanner struct {
-	pluginPath           string
-	errors               []error
-	backendPluginManager backendplugin.Manager
-	cfg                  *setting.Cfg
-	requireSigned        bool
-	log                  log.Logger
-	plugins              map[string]*PluginBase
+	pluginPath                    string
+	errors                        []error
+	backendPluginManager          backendplugin.Manager
+	cfg                           *setting.Cfg
+	requireSigned                 bool
+	log                           log.Logger
+	plugins                       map[string]*PluginBase
+	allowUnsignedPluginsCondition func(plugin *PluginBase) bool
 }
 
 type PluginManager struct {
-	BackendPluginManager backendplugin.Manager `inject:""`
-	Cfg                  *setting.Cfg          `inject:""`
-	log                  log.Logger
-	scanningErrors       []error
+	BackendPluginManager          backendplugin.Manager `inject:""`
+	Cfg                           *setting.Cfg          `inject:""`
+	log                           log.Logger
+	scanningErrors                []error
+	allowUnsignedPluginsCondition func(plugin *PluginBase) bool
 }
 
 func init() {
 	registry.RegisterService(&PluginManager{})
 }
 
+// ReplaceAllowUnsignedPluginCondition has to be called before PluginManager.Init() has run
+func (pm *PluginManager) ReplaceAllowUnsignedPluginCondition(condition func(base *PluginBase) bool) {
+	pm.allowUnsignedPluginsCondition = condition
```

Comment:
🤔 Why is this a method rather than just a public instance variable? Do we expect this setter to get validation logic or something like that?

Reply:
I was also wondering tbh.

Reply:
Probably just my java roots showing. I don't have a good reason for keeping this and will remove it.